bin/create-release

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env ruby
+require_relative '../deployment.rb'
+require 'cdo/only_one'
+require 'cdo/cdo_cli'
+include CdoCli
+
+require 'octokit'
+require 'open3'
+require 'tempfile'
+
+REPO = 'Hamms/misc'
+
+def git_generate_changelog
+  changes, _err, status = Open3.capture3("git log --merges --oneline production^1..production^2 --format=\"%s %b\"")
+
+  # command will error out unless the latest commit is a merge
+  return nil if status.exitstatus != 0
+
+  re = /Merge pull request #(?<pr>\d*) from code-dot-org\/(?<branch>\S*) (?<description>.*)/
+
+  matches = changes.split("\n").map do |change|
+    # first, extract the relevant data from the commit message
+    re.match(change)
+  end
+
+  relevant_matches = matches.select do |match|
+    # ignore merges from staging; those represent DTTs
+    match['branch'] != 'staging'
+  end
+
+  items = relevant_matches.map do |match|
+    # finally, create a changelog entry linking the description to the PR
+    "- [#{match['description']}](https://github.com/#{REPO}/pull/#{match['pr']})"
+  end
+
+  items.join("\n")
+end
+
+def get_next_release_version
+  # version format is "v{year}-{month}-{day}.{version number}"
+  today = Time.now.strftime("v%Y-%m-%d")
+  version_number = 0
+
+  # if we've already released a .0 or more today, increment the version
+  # number appropriately
+  last_version = Octokit.latest_release(REPO).tag_name
+  if last_version.start_with?(today)
+    last_release_version = /v\d{4}-\d{2}-\d{2}\.(\d*)/.match(last_version)[0].to_i
```

Comment:
Validating the version number is probably worth doing, otherwise the release creation step will fail horribly. Since we expect this to happen fairly rarely, I think it's fine to do a somewhat "expensive" solution like keep incrementing the version number and checking to see if it exists (making an API request each time) until we get one that doesn't.